base/src/types.rs

```diff
@@ -106,8 +106,13 @@ pub enum Type<Id, T = AstType<Id>> {
     Record {
         /// The associated types of this record type
         types: Vec<Field<Id, Alias<Id, T>>>,
+        row: T,
+    },
+    EmptyRow,
```

Comment:
> I think there is a TRecord Type variant in AlgorithmW_Records because you do need a record constructor. So we would need to switch to Type::Record(T).

That's already how the current representation is, only with 'type fields' packed into `Type::Record` as well since I didn't want to deal with polymorphic 'type fields' just yet.

> Another idea would be to have record construction be implicit, like Elm does.
Iit might be that the [TermN](https://github.com/elm-lang/elm-compiler/blob/master/src/Type/Type.hs#L54) constructor serves as `TRecord Type`. Without `TRecord : Row -> Type` I am not sure how to reconcile kind checking as with just say `EmptyRecord` and `ExtendRecord` they may have the kind `Row` or `Type` depending on context.